### ISSUE TITLE
Disk attach/detach

### ISSUE THREAD

**Issue by crawlingChaos** [joncampbell123/dosbox-x#189]
I realize disk mounting was inherited from DOSBox. DOSBox's method of mounting all needed disk images before booting into another OS may be fine for the limited scenarios it is designed for, but as a general solution for Win9x it is completely inadequate.

If you have a floppy disk mounted, it becomes impossible to swap between cd-roms because ctrl-F4 will only swap floppies at that point.

Being able to attach/detach disk images to a running emulator is such a common scenario (and generally expected among emulator users) that it would be crazy to not support this. A management UI available while running the emulation ala MAME/MESS would be a nice approach. I honestly have no idea how much work this would be to implement, but am willing to take a look.

**Comment by Jarvik7** [comment]
Definietly this. I think this is the biggest usability issue with DosBox(-x) at present.
Originally DosBox was meant to run things in its DosBox simulation layer which made this a non-issue, but it seems like most -X users prefer to boot into a real OS.

**Comment by joncampbell123** [comment]
I agree. DOSBox-X needs the ability to load ANY floppy disk or ISO image at any time, not what you pre-defined at boot.

I'm thinking an alternate console or overlay to type in commands QEMU-style. Would that work?

**Comment by Jarvik7** [comment]
I've never actually used QEMU (oddly), but something like the mame interface would be nice to have for not just mounting, but control mapping and settings in general. Another less user friendly alternative could be a quake-like console.

**Comment by crawlingChaos** [comment]
One thing I like about MAME (now with MESS integrated) is that for emulation of computer systems, there are two keyboard emulation modes, partial and full.

In full mode, key mapping conflicts are eliminated by disabling UI keys and sending all keys to the emulated system.

In partial mode, some keys are handled by the emulator to control pausing, UI menus, disk swapping, etc.

Switching between modes is performed by tapping the Scroll Lock key: **perhaps the least used key known to mankind**.

No idea what you'd do if you needed to use Scroll Lock in the emulation...but I can't imagine it has ever come up.

**Comment by sikthehedgehog** [comment]
The problem is that Scroll Lock is missing in just about every single laptop out there.

Possibly more relevant problem: it was way more common back in the day if you go outside gaming, since its original purpose was to switch between using arrows to move the cursor and to scroll. (these days we scroll by trying to move the cursor outbounds instead) This is why Excel supports that behavior, incidentally. You'd want a chorded combination here instead.

**Comment by crawlingChaos** [comment]
MAME defaults to Scroll Lock, but you can remap it to either a single key or a combo. I wouldn't want to hardcode any particular combo because no matter what you pick, it is going to cause a problem with *something*.
